c++/src/kj/common.h

```diff
@@ -1753,13 +1753,14 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
     return other.size_ <= size_ && slice(0, other.size_) == other;
   }
 
-  inline Maybe<size_t> findFirst(char c) const {
-    const char* pos = reinterpret_cast<const char*>(memchr(ptr, c, size_));
-    if (pos == nullptr) {
-      return nullptr;
-    } else {
-      return pos - ptr;
+  inline Maybe<size_t> findFirst(const T& match) const {
+    for (int i = 0; i < size_; i++) {
+      // TODO(now): Do we need to consider if `ptr + i` == nullptr?
```

Comment:
How could that be?

Reply:
https://github.com/capnproto/capnproto/pull/1480/commits/8a731e27692bcd083d7a4e6052789b23a0db99df

It's been a while, got a bit rusty so I had to write stuff out :grin:. My main concern was a situation where the elements themselves were pointers, so `ArrayPtr<int*>`, with the array of pointers `{0x100, 0x104, nullptr, 0x112}`. I forgot that the underlying `ptr` of `ArrayPtr<T>` is a `T*`, so it would be an `int **`, and `*(ptr + 2) == nullptr` (I was concerned we'd by mistake dereference the `nullptr`).